### PR TITLE
EfilingClient

### PR DESCRIPTION
**Pull request by KFloodBCGov**
# Description

This PR includes the following proposed change(s):

Efiling Client and Service for calling into CSO SOAP endpoint.
WSDL parsing for SOAP interface to consume.
Renamed the "efilling" folder to "efiling"

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Simple local unit test.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/libs/efiling-facade-client/pom.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    <schemaDirectory>${project.basedir}/src/main/resources/wsdl</schemaDirectory>
+                    <generatePackage>ca.bc.gov.open.jag.ag.csows.filing</generatePackage>
+                    <schemaIncludes>
+                        <schemaIncludes>FilingFacade.wsdl</schemaIncludes>
+                    </schemaIncludes>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-facade-client/pom.xml:69]: #### Nit
newline missing

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/EfilingFacadeClient.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.math.BigDecimal;
+
+
+public class EfilingFacadeClient  {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EfilingFacadeClient.class);
+    private static final EfilingFacadeService efilingFacadeService = new EfilingFacadeService();
+
+    public BigDecimal SubmitFiling(String strFileInfo/* This will require some real data */) {
```
Comment (akroon3r) [inline src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/EfilingFacadeClient.java:15]: slap a TODO here perhaps?

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+            LOGGER.info("Calling SubmitFiling with package id: " + filingPackage.getApplicationReferenceGuid());
+            SubmitFilingResponse response =
+                    (SubmitFilingResponse) getWebServiceTemplate().marshalSendAndReceive(request);
+            return response.getReturn();
+
+        } catch (Exception e) {
+
+            LOGGER.error("Error calling SubmitFiling: " + e.toString());
+            return BigDecimal.ZERO;
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java:41]: #### Question
Why are we returning zero on fail?

Comment (akroon3r) [inline src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java:41]: why not return the exception here?

File: src/backend/pom.xml [hunk 4]
```diff
@@ -23,14 +23,14 @@
 		<profile>
 			<id>all</id>
 			<modules>
-				<module>libs/efilling-facade-client</module>
+				<module>libs/efiling-facade-client</module>
```
Comment (akroon3r) [inline src/backend/pom.xml:26]: 🚢

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/config/EfilingFacadeConfig.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    public Jaxb2Marshaller marshaller(){
+        Jaxb2Marshaller jaxb2Marshaller = new Jaxb2Marshaller();
+        jaxb2Marshaller.setContextPath("ca.bc.gov.open.jag.ag.csows.filing");
+        return jaxb2Marshaller;
+    }
+
+    @Bean
+    public EfilingFacadeService eFilingFacadeClient(Jaxb2Marshaller jaxb2Marshaller) {
+        EfilingFacadeService eFilingFacadeService = new EfilingFacadeService();
+        eFilingFacadeService.setDefaultUri("http://wsgw.dev.jag.gov.bc.ca:8080/csowsJBOSS/FilingFacade");
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/config/EfilingFacadeConfig.java:21]: #### Question
should this url be an env var?

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java (hunk 3 above)
Comment (akroon3r) [reply src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java:41]: We both picked up on that - nice

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+@Service
+public class EfilingFacadeService extends WebServiceGatewaySupport {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EfilingFacadeService.class);
+
+    public BigDecimal SubmitFiling(String strFileInfo) {
```
Comment (alexjoybc) [inline src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java:19]: should that not be a `void` and just let it throw `runtimeException` ?

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java (hunk 3 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java:41]: Just doing that until we get some error handling in place?

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java (hunk 6 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java:19]: The SubmitFiling SOAP call actually returns a BigDecimal.

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/EfilingFacadeClient.java (hunk 2 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/EfilingFacadeClient.java]: Done!

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java (hunk 3 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/EfilingFacadeService.java:40]: I did it like that almost strictly for easy testing until I have some idea of how we want to handle an error.

File: src/backend/libs/efiling-facade-client/pom.xml (hunk 1 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-facade-client/pom.xml:72]: Think I fixed it

File: src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/config/EfilingFacadeConfig.java (hunk 5 above)
Comment (KFloodBCGov) [reply src/backend/libs/efiling-facade-client/src/main/java/ca/bc/gov/open/jag/efilingfacadeclient/service/config/EfilingFacadeConfig.java]: Moved it  to env variable and added properties to config

**Pull request merged**